File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_to_index = {}
        for i, num in enumerate(nums):
            if num_to_index.get(target - num) is not None:
                return [num_to_index[target - num], i];
            num_to_index[num] = i

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const numsSet = new Set(nums);
        let maxLen = 0;
        for (let num of nums) {
            if (!numsSet.has(num - 1)) {
                let length = 1;
                while (numsSet.has(num + length)) ++length;
                maxLen = Math.max(maxLen, length);
            }
        }
        return maxLen;
    }
}
